### ISSUE TITLE
Discussion on the future of schema generators

### ISSUE THREAD

**Issue by HoldYourWaffle** [YousefED/typescript-json-schema#295]
There's currently some discussion on the future of schema generators over at https://github.com/vega/ts-json-schema-generator/issues/101.

It's mostly related to uniting the 3 big (and unfortunately buggy) modules that are used for the same purpose (generating JSON schema's from TypeScript sources) together into one, (mostly) feature complete all-round generator.

These 3 modules are:
- `typescript-json-schema` (this repository)
- `typescript-to-json-schema` at [xiag-ag/typescript-to-json-schemas](https://github.com/xiag-ag/typescript-to-json-schema)
- `ts-json-schema-generator` at [vega/ts-json-schema-generator](https://github.com/vega/ts-json-schema-generator)

We're currently thinking about deprecating this module in favor of [vega/ts-json-schema-generator](https://github.com/vega/ts-json-schema-generator) somewhere in the future (for various reasons you can read in the aforementioned issue). However, nothing has really been decided yet and we would love to hear some more opinions on this matter!

**Comment by JavierPAYTEF** [comment]
It would be good to point somewhere because there are a lot of pending pull requests and issues. I was about to ask about supporting bigint but I see a lot of people mentioning not supporting Symbol with no answers, and most of the issues don't seem to have any answers either. Is this still active?

**Comment by alfaproject** [comment]
If I'm starting a new project that requires this functionality, should I prefer this library, or would you recommend another one?

**Comment by domoritz** [comment]
vega/ts-json-schema-generator is a lot better than this one on edge cases. 

I'm curious why you want to start a new project. Maybe you can help with the new one I linked to.

**Comment by domoritz** [comment]
Yes. This repo is in maintenance mode and I personally don't fix anything and only accept prs and make releases.

**Comment by bdarcus** [comment]
FWIW, I am working on a project where I started with this package for schema generation, then moved to `ts-json-schema-generator` based on this issue, where I triggered a bug there on a feature branch that this package avoids.

**Comment by tomwidmer** [comment]
quicktype appears to use [@mark.probst/typescript-json-schema](https://www.npmjs.com/package/@mark.probst/typescript-json-schema), which is a mysterious fork of this repo - I don't know where the source code lives.

**Comment by domoritz** [comment]
Odd. I wonder why they don't use this fork.
